### PR TITLE
feat(issuer): [SFEQS-1594] Add input documents metadata in CreateSignatureRequest

### PR DESCRIPTION
**Pull request by silvicir**
#### List of Changes

1. Added optional input `documents_metadata` for `CreateSignatureRequest` function on `io-sign-func-issuer` workspace
2. Added unit tests

#### Motivation and Context

The issuer could now set specific signature fields for a given signature request to override dossier's ones

#### How Has This Been Tested?

Unit test + manual e2e test in development environment

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by grausof — APPROVED** [review]
LGTM

**Comment by grausof** [comment]
Add also changeset (`yarn changeset`)

File: apps/io-func-sign-issuer/src/signature-request.ts [hunk 1]
```diff
@@ -57,7 +59,8 @@ export const defaultExpiryDate = () => pipe(new Date(), addDays(90));
 export const newSignatureRequest = (
   dossier: Dossier,
   signer: Signer,
-  issuer: Issuer
+  issuer: Issuer,
+  documentsMetadata: O.Option<NonEmptyArray<DocumentMetadata>>
```
[the remark this answers is not in the document]
Comment (silvicir) [reply apps/io-func-sign-issuer/src/signature-request.ts:63]: I imported `DocumentMetadata` from `@io-sign/io-sign/document` for consistency with the dossier function https://github.com/pagopa/io-sign/blob/main/apps/io-func-sign-issuer/src/dossier.ts#L15

[the remark this answers is not in the document]
Comment (gunzip) [reply apps/io-func-sign-issuer/src/signature-request.ts:63]: data structures should not contain monadic types (ie. Option)

f it's an optional field just use

```suggestion
  documentsMetadata?: NonEmptyArray<DocumentMetadata>
```

This let the payload to be validated using io-ts for example and avoid to navigate the structures using monadic constructors.

**Pull request merged**